src/mame/bandai/wswan.cpp

```diff
@@ -549,27 +514,53 @@ u16 wswan_state::port_r(offs_t offset, u16 mem_mask)
 	switch (offset)
 	{
 		case 0x40 / 2:  // DMA source address
-			return m_dma_source_offset;
+			if (m_vdp->color_mode())
```

Comment:
I gather this is a full on `memory_view` applied to an address map, particularly if the `m_color_mode` is settable from the VDP (that would require a `devcb_write_line`).